### ISSUE TITLE
Test more pack sizes in the AT and/or nightlies.

### ISSUE THREAD

**Issue by bartgol** [E3SM-Project/scream#918]
I tried to run `shoc_standalone` through valgrind, when building with pack size 8. It passed, but raised new valgrind errors (invalid read). Given that the test passed, I speculate it is a harmless error, but nevertheless, using a single pack size could hide bugs.

To be fair, we do test pack size 1 and 16, but they do not cover the whole span of scenarios. More precisely, we want to test the following;

a1) `npacks(nlev)==npacks(nlev+1)
a2) `npacks(nlev)<npacks(nlev+1)`.
b1) pack_size==1
b2) pack_size>1

With pack sizes 16 (dbg and sp builds, in test-all-scream) and 1 (fpe build), and a test with 72 levs, we test a1+b2 and a2+b1, but the other two combinations are not tested. Developing Hommexx, we realized that there are _several_ bugs that can sneak in when we don't stress test the pb.

It might seem like there are 4 combinations of the above options, but it's clear that a1+b1 is not possible. But if we tested, say, pack size=8 with nlev=72, we'd also test a2+b2.

Long story short, when out pack size>1, for a given test we are only testing either a1 or a2, but that might hide some bugs. We need to test both.

**Comment by tcclevenger** [comment]
Yes, I completely agree. Writing tests for EKAT stuff, using both a1 and a2, often found hidden bugs in whatever I was implementing that might have never been caught in SCREAM.

This also shouldn't be more taxing on GPU (since it would only add pack_size>1 tests, right?) and it seems like weaver time is the most valuable these days. So I think it would be useful to have in the AT. How much more time would an extra build cost?

Let me know if you want any help debugging (assuming the tests don't just work out of the box).

**Comment by bartgol** [comment]
On GPU there should not be any added test, since we only support pack_size=1. On CPU, however, there are a few options.
